### PR TITLE
Use reviewdog/action-misspell to find spelling mistakes 

### PR DESCRIPTION
**Pull request by dansman805**
This will automatically find and fix spelling issues and then send in a pr; it happens on push, and can be manually run.
This is what it looks like on my fork of frc-docs: https://github.com/dansman805/frc-docs/pull/1/files.
It caught and fixed 4 spelling errors in the current version of frc-docs (as of the creation of this pr).

### CODE REVIEW

**Comment by Daltz333** [comment]
Ideally I'd also like to see a check, that way we can catch spelling mistakes before their merged.

**Comment by dansman805** [comment]
That'd be possible through a seperate action, this uses the same backend tool but can output as a check
https://github.com/reviewdog/action-misspell

**Comment by dansman805** [comment]
See the misspell report for that latest push to my fork https://github.com/dansman805/frc-docs/runs/1018759185.

**Comment by sciencewhiz** [comment]
Is the reviewdog version using more lists then misspell-fixer was?

**Comment by dansman805** [comment]
It has the same lists, however it has more detection options turned on by default, as at most it can have a false positive that a human notices as opposed to a false positive that gets bundled in a pr where it has greater potential to be accidentally unnoticed.

File: .github/workflows/CI.yaml [hunk 1]
```diff
@@ -80,3 +80,15 @@ jobs:
     - name: Check Image Size
       run: |
         make sizecheck
+  check-spelling:
+    name: Check Spelling
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code.
+        uses: actions/checkout@v1
```
Comment (Daltz333) [inline .github/workflows/CI.yaml:88]: Can we be consistent on the checkout with the other checks?

File: .github/workflows/CI.yaml [hunk 2]
```diff
@@ -80,3 +80,15 @@ jobs:
...
+      - name: misspell
+        uses: reviewdog/action-misspell@v1
+        with:
+          github_token: ${{ secrets.github_token }}
+          locale: "US"
+          reporter: "github-check"
```
Comment (Daltz333) [inline .github/workflows/CI.yaml:94]: Newline

File: .github/workflows/CI.yaml [hunk 3]
```diff
@@ -80,3 +80,15 @@ jobs:
...
+  check-spelling:
+    name: Check Spelling
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code.
+        uses: actions/checkout@v1
+      - name: misspell
+        uses: reviewdog/action-misspell@v1
+        with:
+          github_token: ${{ secrets.github_token }}
```
Comment (AustinShalit) [inline .github/workflows/CI.yaml:92]: We do not need to provide a token here, `github.token` is used by default and should be sufficient.

File: source/docs/software/driverstation/driver-station-errors-warnings.rst [hunk 4]
```diff
@@ -52,11 +52,11 @@ Radio Detection Times
 
 .. code-block::
 
-  WARNING<Code>44008 occured at FRC_NetworkCommunications
+  WARNING<Code>44008 occurred at FRC_NetworkCommunications
```
Comment (AustinShalit) [inline source/docs/software/driverstation/driver-station-errors-warnings.rst:55]: Is this misspelled in the actual error? If so then we need to leave it and ask NI to update the spelling on their end.

File: .github/workflows/CI.yaml (hunk 1 above)
Comment (AustinShalit) [inline .github/workflows/CI.yaml:88]:
```suggestion
      - uses: actions/checkout@v1
```

File: .github/workflows/CI.yaml (hunk 2 above)
Comment (AustinShalit) [reply .github/workflows/CI.yaml:94]:
```suggestion
          reporter: "github-check"

```

File: source/docs/software/driverstation/driver-station-errors-warnings.rst (hunk 4 above)
Comment (dansman805) [reply source/docs/software/driverstation/driver-station-errors-warnings.rst:55]: I do not know, and as an FTC student I unfortunately can't check easily. Would it be possible for someone else to check?

Comment (AustinShalit) [reply source/docs/software/driverstation/driver-station-errors-warnings.rst:55]: I can take a look later this week unless someone beats me to it.

File: .github/workflows/CI.yaml (hunk 1 above)
Comment (Daltz333) [reply .github/workflows/CI.yaml]: We should use v2 not v1

Comment (dansman805) [reply .github/workflows/CI.yaml]: Fixed

File: source/docs/software/driverstation/driver-station-errors-warnings.rst (hunk 4 above)
Comment (sciencewhiz) [reply source/docs/software/driverstation/driver-station-errors-warnings.rst:55]: Someone quoted it in a CD post with the correct spelling, so either they ran spell check on it, or NI fixed it

**Pull request merged**